### PR TITLE
add shareable APPLIED and INVALIDATED implementations of Result

### PR DESCRIPTION
**Pull request by maedhroz**
patch by Caleb Rackliffe; reviewed by ? for CASSANDRA-18355

### CODE REVIEW

**Comment by maedhroz** [comment]
https://app.circleci.com/pipelines/github/maedhroz/cassandra?branch=CASSANDRA-18355

File: accord-core/src/main/java/accord/api/Write.java [hunk 1]
```diff
@@ -30,5 +31,5 @@
  */
 public interface Write
 {
-    AsyncChain<Void> apply(Seekable key, SafeCommandStore safeStore, Timestamp executeAt, DataStore store);
+    AsyncChain<Void> apply(Seekable key, SafeCommandStore safeStore, Timestamp executeAt, DataStore store, Command.Executed command);
```
Comment (belliottsmith) [inline accord-core/src/main/java/accord/api/Write.java]: Do we need to provide the whole command, or could we provide just the `PartialTxn` or `Update`?

Comment (dcapwell) [inline accord-core/src/main/java/accord/api/Write.java]: I kinda don't like exposing `Command` in the API and kinda prefer `TxnId`... you do the following in C*

```
      // Pull updates for this key that did not require completion form the transaction itself and apply them: 
            TxnUpdate txnUpdate = (TxnUpdate) command.partialTxn().update();
            assert txnUpdate != null : "PartialTxn should contain an update if we're applying a write!";
            List<Update> updates = txnUpdate.completeUpdatesForKey((RoutableKey) key);
            updates.forEach(update -> results.add(update.write(timestamp, nowInSeconds)));
```

so you will need it, but I just kinda don't like it...

Comment (maedhroz) [reply accord-core/src/main/java/accord/api/Write.java]: @belliottsmith above suggests using just the `PartialTxn` or the `Update`. You're suggesting perhaps passing `TxnId` and getting the command from `SafeCommandStore` (along w/ the key, which I think we also have available)?

**Review by aweisberg — APPROVED** [review]

**Review by dcapwell — APPROVED** [review]

**Pull request merged**